cdap-watchdog/src/main/java/io/cdap/cdap/logging/clean/LogFolderCleaner.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.logging.clean;
+
+import org.apache.twill.filesystem.Location;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+
+/**
+ * LogFolderCleaner is responsible for cleaning up accumulated empty logs folders.
+ */
+public class LogFolderCleaner implements Runnable {
+
+    /**
+     * FolderCleanupFileVisitor traverses the file tree in Depth-First manner.
+     * Cleanups up empty folders.
+     * @param <T>
+     */
+    private class FolderCleanupFileVisitor<T extends Path> extends SimpleFileVisitor<Path> {
```

Comment:
make it static. It often helps not to have references to parent unless really needed